test/test-generate-pras.jl

```diff
@@ -0,0 +1,241 @@
+    idx = 1
+    while idx <= length(x) && idx <= length(y)
+        if x[idx] == y[idx]
+            idx += 1
+        elseif x[idx] != y[idx]
+            prev_idx = max(idx - 2, 1)
+            next_idx = min(idx + 2, length(x))
+            @error "$left $(x[prev_idx:next_idx]) != $right $(y[prev_idx:next_idx])"
+            return false
+        end
+    end
+    if idx > length(x) && idx > length(y)
+        return true
+    end
+    prev_idx = max(idx - 2, 1)
+    next_idx = min(idx + 2, length(x))
+    comparison = idx <= length(x) ? "more" : "less"
+    @error "$left has $comparison elements $(x[prev_idx:next_idx]) than $right $(y[prev_idx:next_idx])"
+    return false
+end
+
+function array_all_equal(x::AbstractVector{T}, v::T) where {T}
+    for (i, xi) in enumerate(x)
+        if !isapprox(xi, v)
+            @error "array[$i]: $xi != $v"
+            return false
+        end
+    end
+    return true
+end
+
+@testset "RTS GMLC" begin
+    rts_da_sys = get_rts_gmlc_outage()
+    area_names = PSY.get_name.(PSY.get_components(PSY.Area, rts_da_sys))
+    generator_names =
+        PSY.get_name.(
+            PSY.get_components(
+                x -> PSY.get_available(x) && PSY.get_rating(x) > 0,
+                Union{PSY.HydroDispatch, PSY.RenewableGen, PSY.ThermalGen},
+                rts_da_sys,
+            )
+        )
+    storage_names = PSY.get_name.(PSY.get_components(PSY.Storage, rts_da_sys))
+    generatorstorage_names =
+        PSY.get_name.(
+            PSY.get_components(
+                x -> PSY.get_available(x) && PSY.get_rating(x) > 0,
+                Union{PSY.HydroEnergyReservoir, PSY.HybridSystem},
+                rts_da_sys,
+            )
+        )
+    line_names =
+        PSY.get_name.(
+            PSY.get_components(PSY.Branch, rts_da_sys) do c
+                PSY.get_available(c) && !(
+                    c isa PSY.TapTransformer ||
+                    c isa PSY.Transformer2W ||
+                    c isa PSY.PhaseShiftingTransformer
+                )  # From definitions.jl
+                PSY.get_area(PSY.get_from_bus(c)) != PSY.get_area(PSY.get_to_bus(c))
+            end
+        )
+
+    # Make a PRAS System from PSY-4.X System
+    rts_pras_sys = generate_pras_system(rts_da_sys, PSY.Area)
+    @test rts_pras_sys isa PRASInterface.PRAS.SystemModel
+
+    @test test_names_equal(rts_pras_sys.regions.names, area_names)
+    @test test_names_equal(rts_pras_sys.generators.names, generator_names)
+    @test test_names_equal(rts_pras_sys.storages.names, storage_names)
+    @test test_names_equal(rts_pras_sys.generatorstorages.names, generatorstorage_names)
+    @test test_names_equal(rts_pras_sys.lines.names, line_names)
+
+    # Test that timestamps look right
+    # get time series length
+    psy_ts = first(PSY.get_time_series_multiple(rts_da_sys, type=PSY.SingleTimeSeries))
+    @test all(
+        TimeSeries.timestamp(psy_ts.data) .== collect(DateTime.(rts_pras_sys.timestamps)),
```

Comment:
Are you doing _DateTime._ to get rid of the localization issues?

Reply:
Yes. PRAS uses TimeZones, but usually these are just set to UTC by default.